src/test/article.spec.js

```diff
@@ -213,3 +220,418 @@ describe('GET api/v1/articles', () => {
+
+    const response = await server()
+      .delete(`${ARTICLES_API}/${validArticle.id}`)
+      .set('x-access-token', 'greatisthyfaithfulness')
+      .send({
+        title: articleSample.title,
+        description: articleSample.description,
+        body: articleSample.body,
+        image: articleSample.image
+      });
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal('Token is not valid');
+  });
+
+  it('should throw a 400 server error when token is not provided', async () => {
+    beforeEach(async () => {
+      await db.Bookmark.destroy({ cascade: true, truncate: true });
+      await db.User.destroy({ cascade: true, truncate: true });
+      await db.Article.destroy({ cascade: true, truncate: true });
+    });
+
+    const validArticle = await BaseRepository.create(db.Article, articleSample);
+
+    const response = await server()
+      .delete(`${ARTICLES_API}/${validArticle.id}`)
+      .send({
+        title: articleSample.title,
+        description: articleSample.description,
+        body: articleSample.body,
+        image: articleSample.image
+      });
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal('Invalid access token');
+  });
+
+  it('should successfully delete an article', async () => {
+    beforeEach(async () => {
+      await db.Bookmark.destroy({ cascade: true, truncate: true });
+      await db.User.destroy({ cascade: true, truncate: true });
+      await db.Article.destroy({ cascade: true, truncate: true });
+    });
+
+    const newUser = await createUser();
+    const token = helper.jwtSigner(newUser);
+
+    const validArticle = await BaseRepository.create(db.Article, articleSample);
+
+    const response = await chai
+      .request(app)
+      .delete(`${ARTICLES_API}/${validArticle.id}`)
+      .set('x-access-token', token);
+    expect(response.status).to.equal(200);
+    expect(response.body.message).to.equal('Article successfully deleted');
+  });
+
+  it('should throw a 404 error when a specific article is not found', async () => {
+    beforeEach(async () => {
+      await db.Bookmark.destroy({ cascade: true, truncate: true });
+      await db.User.destroy({ cascade: true, truncate: true });
+      await db.Article.destroy({ cascade: true, truncate: true });
+    });
+    const newUser = await createUser();
+    const token = helper.jwtSigner(newUser);
+
+    const validArticle = await BaseRepository.create(db.Article, articleSample);
+
+    const response = await chai
+      .request(app)
+      .get(`${ARTICLES_API}/000010000100`)
+      .set('x-access-token', token)
+      .send(!validArticle);
+    expect(response.status).to.equal(404);
+    expect(response.body.message).to.equal(
+      'The requested article was not found'
+    );
+  });
+
+  it('should throw a 404 server error when searching for an article with a string as articleId', async () => {
```

Comment:
This is a case of bad data. It should be `400`.
`it('should throw a 400 error if article Id is invalid or non-zero positive integer', ...`